File: study/src/main/java/com/sharevar/study/data/server/RemoteDataSource.java
package com.sharevar.study.data.server;

import com.sharevar.study.entity.Course;
import com.sharevar.study.entity.ResponseData;

import java.io.IOException;
import java.util.List;

public class RemoteDataSource {
    private APIService service;
    private static RemoteDataSource instance=new RemoteDataSource();

    public static RemoteDataSource getInstance() {
        return instance;
    }

    public RemoteDataSource() {
        service=RetrofitProvider.create().create(APIService.class);
    }

    public ResponseData<List<Course>> courseList(){
        try {
            return service.courseList().execute().body();
        } catch (IOException e) {
            e.printStackTrace();
            return new ResponseData<>();
        }
    }

    public ResponseData<String> fileContent(String path){
        try {
            return service.fileContent(path).execute().body();
        } catch (IOException e) {
            e.printStackTrace();
            return new ResponseData<>();
        }
    }
}

File: miquan/src/main/java/com/example/miquan/data/server/RemoteDataSource.kt
package com.example.miquan.data.server

import com.example.miquan.model.Price
import com.example.miquan.model.Resource
import com.example.miquan.model.ResourceDetail
import com.example.miquan.model.User
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class RemoteDataSource {
    companion object {
        val instance = RemoteDataSource()
    }

    val service = RetrofitProvider.create().create(ApiService::class.java)

    init {
    }

    fun resouces(page: Int, token: String = "", limit: Int = 10): List<Resource>? {
        val response = service.resource(page, token, limit).execute()
        return response.body()
    }

    fun resourceDetail(id: Int, token: String = ""): ResourceDetail? {
        return service.resourceDetail(id, token).execute().body()
    }

    fun smsCode(phone: String): Boolean? {
        return service.getCode(phone).execute().body()
    }

    fun login(phone: String, code: String): User? {
        return service.login(phone, code).execute().body()
    }

    fun charge(token: String, type: Int): String? {
        return service.charge(token, type).execute().body()
    }

    fun priceConfig(): Price? {
        return service.priceConfig().execute().body()
    }
}